cmake-modules/gcc-arm-toolchain.cmake

```diff
@@ -0,0 +1,5 @@
+set(CMAKE_SYSTEM_NAME Generic)
+set(CMAKE_SYSTEM_PROCESSOR arm)
+set(CMAKE_C_COMPILER arm-none-eabi-gcc)
+set(CMAKE_CXX_COMPILER arm-none-eabi-g++)
+string(APPEND CMAKE_C_FLAGS_INIT " -mcpu=cortex-m4 -ffreestanding -nostdlib -mthumb")
```

Comment:
Can we add some comments to what all these flags being passed in mean? Like `-mthumb`, `-ffreestanding`?